### PR TITLE
Update changelog: Dirty's *_changed? now boolean, never nil

### PR DESCRIPTION
**Pull request by jaredbeck**
See discussion in https://github.com/rails/rails/pull/24515

### CODE REVIEW

**Comment by jaredbeck** [comment]
r? @rafaelfranca 

Rafael is familiar with discussion in https://github.com/rails/rails/pull/24515

**Comment by jaredbeck** [comment]
> Pull requests are usually filed against the master branch for this repo, but this one is against 5-0-stable. Please double check that you specified the right target!

Thanks, I think it's the right branch. I don't know why rails does changelogs this way .. seems more complicated than necessary .. but I think this is the right branch.

**Comment by maclover7** [comment]
@jaredbeck Thank you for opening this PR! Can you open this against master instead of 5-0-stable? We can then backport it to 5-0-stable from there.

**Comment by jaredbeck** [comment]
> Can you open this against master instead of 5-0-stable? We can then backport it to 5-0-stable from there.

Are you sure? I'm confused. The master branch specifically says "Please check 5-0-stable for previous changes." ie. the 5.0 changelog can only be found in the 5.0-stable branch, right? (https://github.com/rails/rails/blob/5-0-stable/activerecord/CHANGELOG.md)

**Comment by rafaelfranca** [comment]
Right.

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,11 @@
 
 ## Rails 5.0.0 (June 30, 2016) ##
 
+*   AM::Dirty's changed? methods now return an actual boolean, never `nil`, as in 4.2.
```
Comment (kaspth) [inline activerecord/CHANGELOG.md]: We should spell out `ActiveModel::Dirty`, wrap `changed?` in backticks and move this to the correct changelog 😁

We also prefer `singleton` to `boolean`. "actual" reads a bit condescending to me, I'd snip that.

You say "methods" but `changed?` is only one, were you referring to the `*_changed?` methods?

File: activerecord/CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,11 @@
...
+    https://github.com/rails/rails/pull/24515
```
Comment (kaspth) [inline activerecord/CHANGELOG.md]: Our style is to not link back to PRs.

Comment (jaredbeck) [inline activerecord/CHANGELOG.md]: Why? A link is very useful to someone researching the reasons behind a change.

File: activerecord/CHANGELOG.md (hunk 1 above)
Comment (jaredbeck) [inline activerecord/CHANGELOG.md]: Thanks for the feedback Kasper. I can make all of those changes, except "move this to the correct changelog" which I don't understand. See comments above about branches.

Comment (rafaelfranca) [inline activerecord/CHANGELOG.md]: No need to worry about those. I already made the changes

File: activerecord/CHANGELOG.md (hunk 2 above)
Comment (kaspth) [inline activerecord/CHANGELOG.md]: True and realized in Rafael's updated comment that "Fixes #" is preferred and has precedent in the changelogs 😁

**Pull request merged**